tests/ui/components/search/searchcomponent.js

```diff
@@ -79,4 +80,23 @@ describe('SearchBar component', () => {
       expect(storage.getUrlWithCurrentState()).toEqual('query=');
     });
   });
+
+  it('default initial search works for universal', () => {
+    const defaultInitialSearch = '';
+    storage.set(StorageKeys.QUERY, defaultInitialSearch);
+    storage.set(StorageKeys.QUERY_TRIGGER, QueryTriggers.INITIALIZE);
+
+    const component = COMPONENT_MANAGER.create('SearchBar', {
+      ...defaultConfig,
+      allowEmptySearch: true
+    });
+
+    const wasSearchRanPromise = new Promise(resolve => {
+      component.search = jest.fn(() => {
```

Comment:
I tried that, but I wasn't able to call the promise resolve function from within this test. Do you have other ideas on how to test that it was called in an asynchronous way?

Reply:
ah I see that IS tricky because there's no callback for the component's async stuff. I think we could still change the mock from component.search to component.core.search though

Reply:
updated!